backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/RDFApi.java

```diff
@@ -16,35 +16,107 @@
 import org.molgenis.emx2.rdf.RDFService;
 import spark.Request;
 import spark.Response;
+import spark.Route;
 import spark.utils.StringUtils;
 
 public class RDFApi {
   public static final String FORMAT = "format";
   private static MolgenisSessionManager sessionManager;
   public static final String RDF_API_LOCATION = "/api/rdf";
+  public static final String TTL_API_LOCATION = "/api/ttl";
+  public static final String JSONLD_API_LOCATION = "/api/jsonld";
+
+  private static final List<RDFFormat> acceptedRdfFormats =
+      List.of(
+          RDFFormat.TURTLE,
+          RDFFormat.N3,
+          RDFFormat.NTRIPLES,
+          RDFFormat.NQUADS,
+          RDFFormat.RDFXML,
+          RDFFormat.TRIG,
+          RDFFormat.JSONLD);
 
   public static void create(MolgenisSessionManager sm) {
     // ideally, we estimate/calculate the content length and inform the client using
     // response.raw().setContentLengthLong(x) but since the output is streaming and the triples
     // created on-the-fly, there is no way of knowing (or is there?)
     sessionManager = sm;
-    get(RDF_API_LOCATION, RDFApi::rdfForDatabase);
-    get("/api/jsonld", RDFApi::jslonldForDatabase);
-    get("/api/ttl", RDFApi::ttlForDatabase);
-    final String schemaPath = "/:schema" + RDF_API_LOCATION;
-    get(schemaPath, RDFApi::rdfForSchema);
-    // FIXME: rdfForTable also handles requests for a specific row if there is a composite key
-    // TODO: probably best to merge these two methods and always use query string to encode the row
-    get(schemaPath + "/:table", RDFApi::rdfForTable);
-    get(schemaPath + "/:table/:row", RDFApi::rdfForRow);
-    get(schemaPath + "/:table/column/:column", RDFApi::rdfForColumn);
-    get("/:schema/api/jsonld", RDFApi::jsonldForSchema);
-    get("/:schema/api/ttl", RDFApi::ttlForSchema);
-    get("/:schema/api/jsonld/:table", RDFApi::jsonldForTable);
-    get("/:schema/api/ttl/:table", RDFApi::ttlForTable);
-  }
-
-  private static int jslonldForDatabase(Request request, Response response) throws IOException {
+
+    defineApiRoutes(
+        RDF_API_LOCATION,
+        RDFApi::rdfHead,
+        RDFApi::rdfForDatabase,
+        RDFApi::rdfForSchema,
+        RDFApi::rdfForTable,
+        RDFApi::rdfForRow,
+        RDFApi::rdfForColumn);
+    defineApiRoutes(
+        TTL_API_LOCATION,
+        RDFApi::ttlHead,
+        RDFApi::ttlForDatabase,
+        RDFApi::ttlForSchema,
+        RDFApi::ttlForTable,
+        RDFApi::ttlForRow,
+        RDFApi::ttlForColumn);
+    defineApiRoutes(
+        JSONLD_API_LOCATION,
+        RDFApi::jsonldHead,
+        RDFApi::jsonldForDatabase,
+        RDFApi::jsonldForSchema,
+        RDFApi::jsonldForTable,
+        RDFApi::jsonldForRow,
+        RDFApi::jsonldForColumn);
+  }
+
```

Comment:
Do you need to define al these handlers here?
Can't you send them to the same function in handle defineApiRoutes with an Type parameter and send them to the same function?:
`private static void defineApiRoutes(
      String apiLocation, RDFFormat rdfFormat)`
And then resolve that in the function handler?

Reply:
Original code used extra functions for ttl/jsonld API so simply kept that to reduce code refactoring, though changed it now to your suggestion. Let me know what you think. Also added extra tests to validate on "if jsonld API would get a different content-type request will it still return jsonld"